artifacts/writer.py

```diff
@@ -0,0 +1,132 @@
+# -*- coding: utf-8 -*-
+"""The artifact writer objects."""
+
+import abc
+import json
+import yaml
+
+class BaseArtifactsWriter(object):
+  """Class that implements the artifacts reader interface."""
+
+  @abc.abstractmethod
+  def _ConvertArtifact(self, artifact):
```

Comment:
make this method name more explict e.g. CopyArtifactDefitionToDict also why is this method needed? Wouldn't it make more sense to add a CopyToDist() method to ArtifactDefinition ?